apps/aecore/lib/aecore/account/tx/spend_tx.ex

```diff
@@ -147,14 +143,11 @@ defmodule Aecore.Account.Tx.SpendTx do
           SpendTx.t(),
           DataTx.t()
         ) :: :ok | {:error, reason()}
-  def preprocess_check(
-        accounts,
-        %{},
-        _block_height,
-        %SpendTx{amount: amount},
-        %DataTx{fee: fee} = data_tx
-      ) do
-    %Account{balance: balance} = AccountStateTree.get(accounts, DataTx.main_sender(data_tx))
+  def preprocess_check(accounts, %{}, _block_height, %SpendTx{amount: amount}, %DataTx{
+        fee: fee,
+        senders: [%Identifier{value: sender}]
```

Comment:
I assume this fails if multiple senders are present

Reply:
Number of senders shuld be checked in `validate` and it's enforced by encoding